KubeArmor/core/kubeUpdate.go

```diff
@@ -800,15 +800,16 @@ func (dm *KubeArmorDaemon) WatchK8sPods() {
 
 				dm.K8sPodsLock.Lock()
 
+				duplicateAdd := false
+
 				if event.Type == "ADDED" {
-					new := true
 					for _, k8spod := range dm.K8sPods {
 						if k8spod.Metadata["namespaceName"] == pod.Metadata["namespaceName"] && k8spod.Metadata["podName"] == pod.Metadata["podName"] {
-							new = false
+							duplicateAdd = true
 							break
 						}
 					}
-					if new {
+					if !duplicateAdd {
 						dm.K8sPods = append(dm.K8sPods, pod)
 					}
```

Comment:
It's probably better to keep the earlier code
```suggestion
					if new {
						dm.K8sPods = append(dm.K8sPods, pod)
					} else {
					        events.Type == "MODIFIED" 
					        // because pod already exists
					}
```

Reply:
Good callout. I've now rewritten the commit to include logic more like that, as well as fix a rare case where an update between watch restarts could get missed in my original changes.